sdk/tables/azure-data-tables/azure/data/tables/aio/_table_client_async.py

```diff
@@ -317,20 +317,27 @@ async def update_entity(
         row_key = entity['RowKey']
         entity = _add_entity_properties(entity)
         try:
+            metadata = None
             if mode is UpdateMode.REPLACE:
-                await self._client.table.update_entity(
+                metadata, _ = await self._client.table.update_entity(
                     table=self.table_name,
                     partition_key=partition_key,
                     row_key=row_key,
                     table_entity_properties=entity,
                     if_match=if_match or if_not_match or "*",
+                    cls=kwargs.pop('cls', _return_headers_and_deserialized),
                     **kwargs)
             elif mode is UpdateMode.MERGE:
-                await self._client.table.merge_entity(table=self.table_name, partition_key=partition_key,
-                                                      row_key=row_key, if_match=if_match or if_not_match or "*",
-                                                      table_entity_properties=entity, **kwargs)
+                metadata, _ = await self._client.table.merge_entity(
+                    table=self.table_name,
+                    partition_key=partition_key,
+                    row_key=row_key,
+                    if_match=if_match or if_not_match or "*",
```

Comment:
hmm pretty sure we shouldn't be using `if_not_match` here - I don't think that header is supported.

Reply:
You are correct, confirmed that with tests in separate PR.